src/index.ts

```diff
@@ -543,8 +549,13 @@ export class Pool extends (EventEmitter as new () => PoolEmitter) {
         this.connectionQueueEventEmitter.emit(`connection_${id}`, client);
       } else if (this.options.idleTimeoutMillis > 0) {
         client.idleTimeoutTimer = setTimeout((): void => {
-          // eslint-disable-next-line no-void
-          void this._removeConnection(client);
+          if (this.connections.length < this.options.minPoolSize) {
+            // eslint-disable-next-line no-void
+            void client.release();
```

Comment:
It looks like this should cause a stack overflow, but since a timeout is being used it creates a completely new stack each time.